agent/command/git.go

```diff
@@ -828,7 +844,11 @@ func (c *gitFetchProject) fetch(ctx context.Context,
 		if err := ctx.Err(); err != nil {
 			return errors.Wrapf(err, "canceled while applying module '%s'", moduleName)
 		}
-		err = c.fetchModuleSource(ctx, conf, logger, jpm, opts.token, p, moduleName)
+		appTokens := map[string]string{}
```

Comment:
Assuming we're using `appTokens` as a cache, can we add a clarifying comment that this is acting as a cache mapping repo owners to GitHub app tokens to reduce the number of app tokens generated?

I guess as a follow-up question, is there enough benefit to caching them vs. the potential issue that the token could expire on later clones? I'm wondering if there are edge cases where git clones are particularly slow like [this one](https://ui.honeycomb.io/mongodb-4b/environments/production/datasets/evergreen-agent/result/f4mjgFopujE/trace/gjMb6XsDpYB?fields[]=c_name&fields[]=c_service.name&span=10a8f70440cdb60d), which took 2 hours to clone one of the modules (but it did succeed eventually). Are GitHub app tokens valid for long enough to cover slow (but working) git clones, or will it expire by the time it's reused for the next module?

Reply:
While it only happens a few times, it seems like mandating fast git clones isn't something we should do. The GitHub app token usage is supposed to be something that's transparent to the end user and should "just work", so absent some explicit configuration they've set (like setting exec timeout to 10 mins or something), they shouldn't be worried about their clones being too slow and then having to make their own workarounds because Evergreen is flaky about cloning.

The 1 hour TTL thing may also be troublesome for the app token in the expansion as well, because the time between when the GitHub app generates the token for `PopulateExpansions` and when the task finally clones could be any amount of time, meaning we could have already used up some of the TTL time and the original app token has already expired. If the GitHub app token can only be used for git.get_project, I would make a follow-up ticket to remove the app token from the expansions entirely (which also gets rid of the security risk of leaking them) and just fetch it dynamically like with the module token.